src/test/java/com/puppycrawl/tools/checkstyle/BaseCheckTestSupport.java

```diff
@@ -19,449 +19,11 @@
 
 package com.puppycrawl.tools.checkstyle;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+/**
+ * @deprecated Checkstyle will not support BaseCheckTestSupport anymore. Use
+ *             {@link BaseTestSupport} instead.
+ */
+@Deprecated
+public class BaseCheckTestSupport extends BaseTestSupport {
```

Comment:
@subkrish 
we need to cheat it some how , may by abstract modifier
of no luck - we need to suppress violation. I am ok with suppression in Deprecated class.